libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1472,13 +1490,30 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context.appendConditionalInvalid();
 
 		if (_operator == Token::Div)
-			m_context << (c_isSigned ? Instruction::SDIV : Instruction::DIV);
+		{
+			if (fixedType)
+			{
+				// TODO: not really tested
+				solUnimplementedAssert(fixedFactor < (u256(1) << (fixedType->numBits() / 2)), "Not yet implemented - large FixedPointType division");
+				// Division would cancel out factor, this gives the numerator an extra factor so it remains in result
+				m_context << fixedFactor
+						  << Instruction::MUL;
+			}
+			m_context << (isSigned ? Instruction::SDIV : Instruction::DIV);
+		}
 		else
-			m_context << (c_isSigned ? Instruction::SMOD : Instruction::MOD);
+			m_context << (isSigned ? Instruction::SMOD : Instruction::MOD);
 		break;
 	}
 	case Token::Exp:
-		m_context << Instruction::EXP;
+		if (intType)
+		{
+			m_context << Instruction::EXP;
+		}
+		else
+		{
```

Comment:
Here, we have to put braces, because the windows compiler has problems with such macros.